python/cuml/linear_model/linear_regression.pyx

```diff
@@ -243,15 +233,19 @@ class LinearRegression(Base):
 
         """
 
+        self._set_output_type(X)
+
         cdef uintptr_t X_ptr, y_ptr
-        X_m, X_ptr, n_rows, self.n_cols, self.dtype = \
-            input_to_dev_array(X, check_dtype=[np.float32, np.float64])
+        X_m, n_rows, self.n_cols, self.dtype = \
+            input_to_cuml_array(X, check_dtype=[np.float32, np.float64])
+        X_ptr = X_m.ptr
 
-        y_m, y_ptr, _, _, _ = \
-            input_to_dev_array(y, check_dtype=self.dtype,
-                               convert_to_dtype=(self.dtype if convert_dtype
-                                                 else None),
-                               check_rows=n_rows, check_cols=1)
+        y_m, _, _, _ = \
```

Comment:
Just a question- this seems like it could eventually be a constructor for cumlArray rather than a separate helper function.

Reply:
I've been debating on that in my head for quite a while, and definitely a near future improvement. I wasn't sure of the best way to design it, but after recent work on MGData, I think I have a good idea of how. Will open an issue for it